chainstate/src/detail/mod.rs

```diff
@@ -307,6 +317,95 @@ impl<S: BlockchainStorage> Chainstate<S> {
         Ok(block)
     }
 
+    pub fn get_token_detail(
```

Comment:
get_token_detail... what does detail mean? I have no freaking idea. Just another english word that could mean "stuff". A function that says detail, and then returns an issuance transaction and a block id in a tuple/pair... and to make this even better, the issuance transaction is in TokenIssuanceTransaction struct that has nothing but the issuance transaction (even though I requested token information to contain the full thing), yet the block id is returned as a pair with that struct.

Because function names mean nothing. Function interfaces are just tools to make the code compile. And usually when we want to return data, we just stack it in a tuple instead of making a meaningful interfaces.

To top all this up, we have all these getter functions here completely skipping the query interface.

I seriously feel like I'm gonna get a stroke. My head is numb. I'm really tired of reviewing this PR. Every other day I find more bad practices, negligence and arbitrary names that mean nothing to their function or interface. I'm out of options. This PR has been stringing on for over two months now. I'm trying my best now to fix this myself because I'm tired of the ping-pong we're playing to get this PR merged, which is only resulting in more time wasted.